File: swift/socketd/socketd/src/socketd/transport/core/CodecReader.swift
//
//  BufferReader.swift
//  socketd
//
//  Created by noear on 2023/12/27.
//

import Foundation

/**
 * 编解码缓冲读
 *
 * @author noear
 * @since 2.0
 */
protocol CodecReader{
    /**
     * 获取 byte
     */
    func getByte() -> Int8;
    
    /**
     * 获取一组 byte
     */
    func getBytes( dst:[Int8],  offset:Int32,  length:Int32);
    
    /**
     * 获取 int
     */
    func getInt() -> Int32;
    
    /**
     * 剩余长度
     */
    func remaining() -> Int32;
    
    /**
     * 当前位置
     */
    func position() -> Int32;
}

File: java/socketd/src/main/java/org/noear/socketd/transport/core/CodecReader.java
package org.noear.socketd.transport.core;

/**
 * 编解码缓冲读
 *
 * @author noear
 * @since 2.0
 */
public interface CodecReader {

    /**
     * 获取 byte
     */
    byte getByte();

    /**
     * 获取一组 byte
     */
    void getBytes(byte[] dst, int offset, int length);

    /**
     * 获取 int
     */
    int getInt();

    /*
     * 预看 byte
     * */
    byte peekByte();

    /**
     * 跳过
     */
    void skipBytes(int length);

    /**
     * 剩余长度
     */
    int remaining();

    /**
     * 当前位置
     */
    int position();
}
